src/st/cat/e_lossoth.c

```diff
@@ -0,0 +1,855 @@
+                                         {.x = FIX(2.5), .y = FIX(-4.0)}};
+// Random intervals to walk towards player when aggro'd
+static s16 walk_timers[] = {32, 64, 80, 96};
+static Point16 death_jacket_flame_positions[3] = {
+    {.x = 4, .y = -16}, {.x = -10, .y = -2}, {.x = 2, .y = 4}};
+
+static void DrawNapalmEffect(Primitive* prim) {
+    s32 primY;
+
+    switch (prim->p3) {
+    case 1:
+        prim->v0 = prim->v1 = 0x58;
+        prim->v2 = prim->v3 = 0x58;
+        PGREY(prim, 0) = PGREY(prim, 1) = PGREY(prim, 2) = PGREY(prim, 3) =
+            0x80;
+        prim->drawMode = DRAW_TPAGE2 | DRAW_TPAGE | DRAW_UNK02 | DRAW_TRANSP;
+        if (g_CurrentEntity->facingLeft) {
+            prim->x0 -= 4;
+            prim->x1 -= 4;
+        } else {
+            prim->x0 += 4;
+            prim->x1 += 4;
+        }
+        prim->p3++;
+        // fallthrough
+    case 2:
+        primY = prim->y2 - prim->y0;
+        primY = (0x24 - primY) / 2;
+        if (primY < 2) {
+            primY = 2;
+        }
+        prim->y0 = prim->y1 -= primY;
+        prim->v2 = prim->v3 += primY;
+        prim->x3 = g_CurrentEntity->posX.i.hi;
+        if (prim->v2 >= 0x7C) {
+            prim->drawMode |= DRAW_COLORS;
+            if (g_CurrentEntity->facingLeft) {
+                prim->x3 = prim->x2 + 0x10;
+            } else {
+                prim->x3 = prim->x2 - 0x10;
+            }
+            prim->v2 = prim->v3 = 0x7C;
+            prim->p3++;
+        }
+        break;
+    case 3:
+        prim->y0 = ++prim->y1;
+        primY = prim->y2 - prim->y0;
+        if (primY < 0x10) {
+            prim->p3 = 4;
+        }
+        break;
+    case 4:
+        if (g_CurrentEntity->facingLeft) {
+            prim->x1--;
+            prim->x3--;
+        } else {
+            prim->x1++;
+            prim->x3++;
+        }
+        if (!PrimDecreaseBrightness(prim, 7)) {
+            prim->p3 = 0;
+            prim->drawMode = DRAW_HIDE | DRAW_UNK02;
+        }
+        break;
+    }
+}
+
+extern EInit g_EInitLossoth;
+void EntityLossoth(Entity* self) {
+    Entity* newEntity;
+    s32 i;
+    u32 var_s2;
+
+    typedef enum Step {
+        INIT = 0,
+        DROP_TO_GROUND = 1,
+        IDLE_WALK = 2,
+        PLAYER_AGGRO = 3,
+        ATTACK_FIREBALLS = 4,
+        ATTACK_NAPALM_FLARE = 5,
+        DEATH = 8
+    };
+
+    if ((self->flags & FLAG_DEAD) && self->step < 8) {
+        PlaySfxPositional(SFX_LOSSOTH_DEATH);
+        SetStep(DEATH);
+    }
+
+    switch (self->step) {
+    case INIT:
+        InitializeEntity(g_EInitLossoth);
+        self->animCurFrame = 1;
+        self->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+
+        // Handles lantern glow effect and some death flame out visuals
+        newEntity = self + 1;
+        CreateEntityFromEntity(E_LOSSOTH_EFFECTS, self, newEntity);
+        newEntity->zPriority = self->zPriority + 1;
+
+        newEntity = self + 2;
+        CreateEntityFromEntity(E_UNK_43, self, newEntity);
```

Comment:
Not certain what this is just yet, it is `func_us_801BB4CC` in a completely different portion of the overlay
I suspect maybe the background shadow effect from the lava but need to decompile it to confirm.